### PR TITLE
[GameBoard] Add moving logic to gameboard

### PR DESCRIPTION
**Pull request by pranathip**
# Summary
Adding logic to support the different moves (left, right, top, bottom) that a user can make on a game board.

Functions:
- `moveLeft`
- `moveRight`
- `moveUp`
- `moveDown`

# Images/videos

### CODE REVIEW

**Comment by pranathip** [comment]
Current dependencies on/for this PR:
* main
  * **PR #12** <a href="https://app.graphite.dev/github/pr/pranathip/pranathi-2048-react-app/12" target="_blank">Graphite</a> 
    * **PR #13** <a href="https://app.graphite.dev/github/pr/pranathip/pranathi-2048-react-app/13" target="_blank">Graphite</a>  👈
      * **PR #14** <a href="https://app.graphite.dev/github/pr/pranathip/pranathi-2048-react-app/14" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/pranathip/pranathi-2048-react-app/13?utm_source=stack-comment).

File: src/GameBoard.js [hunk 1]
```diff
@@ -52,4 +52,108 @@ const compress = (board) => {
...
+  return board;
+};
+
+export const moveLeft = (board) => {
+  const newBoard1 = compress(board);
+  const newBoard2 = merge(newBoard1);
+  return compress(newBoard2);
+};
+
+const reverse = (board) => {
```
Comment (pranasaurus-rex) [inline src/GameBoard.js:76]: This looks great!

**Review by pranasaurus-rex — APPROVED** [review]
Looks great!

**Comment by pranathip** [comment]
@pranathip started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/pranathip/pranathi-2048-react-app/13).

**Comment by pranathip** [comment]
@pranathip merged this pull request with [Graphite](https://app.graphite.dev/github/pr/pranathip/pranathi-2048-react-app/13).

**Pull request merged**